apps/site/src/pages/account/api-keys-panel/types.ts

```diff
@@ -0,0 +1,34 @@
+import { Dispatch, ReactElement, SetStateAction } from "react";
+
+import { UserFacingApiKeyProperties } from "../../../lib/api/model/api-key.model";
+
+export type ApiKeyProps = UserFacingApiKeyProperties;
+
+export type KeyAction = "rename" | "revoke";
+
+export type KeyActionStatus =
+  | {
+      publicId: string;
+      action: KeyAction;
+    }
+  | undefined;
+
+export interface ApiKeyItemProps {
+  renameApiKeyCard: ReactElement;
+  revokeApiKeyCard: ReactElement;
```

Comment:
I believe `ReactNode` is probably more versitile/consistent.

Reply:
But we're actually expecting a `ReactElement` here, not a `ReactNode`. these two has their corresponding components, so anything else than a `ReactElement` is not expected.

Reply:
The component just renders whatever you pass. There's no reason not to use ReactNode.